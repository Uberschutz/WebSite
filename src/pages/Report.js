import React, { Component } from 'react';
import '../styles/bootstrap.css';
import '../styles/Report.css';

import {ButtonDropdown, DropdownMenu, DropdownToggle} from "reactstrap";
import ReactApexChart from "react-apexcharts";
import {ProgressBar} from "react-bootstrap";

import bad from '../assets/icons8-triste-80.png'
import neutral from '../assets/icons8-neutre-80.png'
import good from '../assets/icons8-content-80.png'

class Report extends Component {

	constructor(props) {
		super(props);

		this.toggle = this.toggle.bind(this);
		this.state = {
			isOpen: false,
			selectedChild: 'Enfants',
			childData: []
		};
	}

	toggle() {
		this.setState({
			isOpen: !this.state.isOpen
		});
	}

	componentDidMount() {
		console.log('whut');
	}

	changeChild(name) {
		const datas = this.getChildData(name);
		this.setState({selectedChild: name, isOpen: false, childData: datas});
	}

	getChildData(name) {
		/* Here call Christian API */
		const min = 0;
		const max = 100;
		return [
			{name: 'Safe', value: min + Math.random() * (max - min)},
			{name: 'Toxicity', value: min + Math.random() * (max - min)},
			{name: 'Obscenity', value: min + Math.random() * (max - min)},
			{name: 'Racism', value: min + Math.random() * (max - min)},
			{name: 'Insult', value: min + Math.random() * (max - min)},
			{name: 'Threat', value: min + Math.random() * (max - min)},
			{name: 'Truly toxic', value: min + Math.random() * (max - min)},
			{name: 'Profanity', value: min + Math.random() * (max - min)},
			{name: 'Inflammatory', value: min + Math.random() * (max - min)},
			{name: 'Identity Attack', value: min + Math.random() * (max - min)},
			{name: 'Hating', value: min + Math.random() * (max - min)},
		];
	}

	render() {
		//if (this.props.logged) {
			//let i = 0;
			return (
				<div>
					<ButtonDropdown isOpen={this.state.isOpen} toggle={this.toggle} size="lg">
						<DropdownToggle color="info">
							{this.state.selectedChild}
						</DropdownToggle>
						<DropdownMenu className="drop btn">
							<div>
								<div onClick={() => {this.changeChild('Thomas')}}>Thomas</div>
							</div>
							<div>
								<div onClick={() => {this.changeChild('Theo')}}>Théo</div>
							</div>
							<div>
								<div onClick={() => {this.changeChild('Philippe')}}>Philippe</div>
							</div>
						</DropdownMenu>
					</ButtonDropdown>
					{/*<BarChart/>*/}
					<div className="uber-color2">
						<span>Blablabla c'est pas bien</span>
					</div>
					<br/>
					<div style={{width: '50%'}} className="btn">
						{
							this.state.childData.map(d => {
								if (d.name === 'Safe') {
									return (
										<div style={{height: 32, margin: 10}}>
											<text style={{float: "left", marginRight: 10, width: 150, textAlign: 'right'}}>{d.name}</text><ProgressBar style={{height: 24, fontSize: 15}} animated striped variant="success" now={Math.round(d.value)} label={`${Math.round(d.value)}%`}/>
										</div>
									);
								} else {
									var color;
									if (d.value <= 33)
										color = "info";
									else if (d.value <= 66)
										color = "warning";
									else
										color = "danger";
									return (
										<div style={{height: 32, margin: 10}}>
											<text style={{float: "left", marginRight: 10, width: 150, textAlign: 'right'}}>{d.name}</text><ProgressBar style={{height: 24, fontSize: 15}} animated striped variant={color} now={Math.round(d.value)} label={`${Math.round(d.value)}%`}/>
										</div>
									);
								}
							})
						}
						{/*<div style={{height: 32, margin: 10}}>
							<text style={{float: "left", marginRight: 10, width: 100, textAlign: 'right'}}>Toxicity</text><ProgressBar style={{height: 24, fontSize: 15}} animated striped variant="success" now={40} label="40%"/>
						</div>
						<div style={{height: 32, margin: 10}}>
							<text style={{float: "left", marginRight: 10, width: 100, textAlign: 'right'}}>Obsenity</text><ProgressBar style={{height: 24}} animated striped variant="info" now={20} label="20%"/>
						</div>
						<div style={{height: 32, margin: 10}}>
							<text style={{float: "left", marginRight: 10, width: 100, textAlign: 'right'}}>Racism</text><ProgressBar style={{height: 24}} animated striped variant="warning" now={60} label="60%"/>
						</div>*/}
					</div>
					<h5>La navigation Internet de votre enfant est à X % saine</h5>
					<Smiley/>
				</div>
			);
		} /*else {
			return (
				<div>
					<Unauthorized/>
				</div>
			)
		}*/
	//}
}

class Smiley extends Component {
	render() {
		return (
			<div>
				<img src={bad} alt="bad"/>
				<img src={neutral} alt="neutral"/>
				<img src={good} alt="good"/>
			</div>
		)
	}
}

class BarChart extends React.Component {

	constructor(props) {
		super(props);

		this.state = {
			options: {
				plotOptions: {
					bar: {
						barHeight: '100%',
						distributed: true,
						horizontal: true,
						dataLabels: {
							position: 'bottom'
						},
					}
				},
				legend: {
					show: false
				},
				colors: ['#33b2df', '#546E7A', '#d4526e', '#13d8aa', '#A5978B', '#2b908f', '#f9a3a4', '#90ee7e'
				],
				dataLabels: {
					enabled: true,
					textAnchor: 'start',
					style: {
						colors: ['#000']
					},
					formatter: function (val, opt) {
						return opt.w.globals.labels[opt.dataPointIndex] + ":  " + val
					},
					offsetX: 0,
					dropShadow: {
						enabled: true
					},
				},
				stroke: {
					width: 1,
					colors: ['#fff']
				},
				xaxis: {
					categories: ['South Korea', 'Canada', 'United Kingdom', 'Netherlands', 'Italy', 'France', 'Japan',
						'United States'
					],
					labels: {
						show: true
					},
					/*axisTicks: {
						show: false
					},*/
					tickAmount: undefined,
					tickPlacement: 'on',
					min: 0,
					max: 100,
					//type: 'numeric'
				},
				yaxis: {
					labels: {
						show: true
					}
				},
				title: {
					text: 'Custom DataLabels',
					align: 'center',
					floating: true,
				},
				subtitle: {
					text: 'Category Names as DataLabels inside bars',
					align: 'center',
				},
				tooltip: {
					theme: 'dark',
					x: {
						show: true
					},
					y: {
						title: {
							formatter: function () {
								return ''
							}
						}
					}
				}
			},
			series: [{
				data: [70, 30, 48, 12, 26, 5, 86, 33]
			}],
		}
	}

	render() {
		return (
			<div id="chart">
				<ReactApexChart options={this.state.options} series={this.state.series} type="bar" height="350"/>
			</div>
		)
	}
}

export default Report;