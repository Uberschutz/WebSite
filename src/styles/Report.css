.uber-color2 {
    background-color: #1d2124;
    color: white!important;
}

.mrg-begin {
    margin-top: 2%;
}

.left-filter {
    float: left;
    height: 100%;
    margin: 1%;
    width: 15%;
}

.filter-align {
    text-align: left;
}

.test {
    margin-left: 0;
    float: left;
    margin-bottom: 2%;
}

.summary {
    clear: both;
}