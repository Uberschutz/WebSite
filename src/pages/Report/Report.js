import React, { Component } from 'react';
import '../../styles/bootstrap.css';
import '../../styles/Report.css';

import {ButtonDropdown, DropdownMenu, DropdownToggle} from "reactstrap";
import {ProgressBar} from "react-bootstrap";
import { displayContent } from '../../utils/translationDisplay';
import axios from 'axios';

import bad from '../../assets/icons8-triste-80.png'
import neutral from '../../assets/icons8-neutre-80.png'
import good from '../../assets/icons8-content-80.png'
import Unauthorized from "../Unauthorized";

class Report extends Component {

	constructor(props) {
		super(props);

		this.toggle = this.toggle.bind(this);
		this.state = {
			isOpen: false,
			selectedChild: undefined,
			childData: [],
			lang: 'fr',
			logged: false,
			token: '',
			childrens: [],
			filters: []
		};
		this.i = 1;
		this.editFilter = this.editFilter.bind(this);
	}

	componentDidMount() {
		if (this.props.base) {
			const { base: { language, logged, token } } = this.props;
				this.setState({
					lang: language,
					logged: logged,
					token
				}, () => {
					axios.post('/children', {
						action: 'list'
					}, {
						headers: {
							'x-access-token': this.state.token
						}
					}).then(response => {
						console.log(response.data);
						this.setState({
							childrens: response.data
						});
					}).catch(err => {
						console.log(err);
					});
				})
		}
	}

	componentDidUpdate(prevProps, prevState, snapshot) {
		// console.log(prevProps, this.props);
		if (this.props.base !== prevProps.base) {
			this.setState({lang: this.props.base.language, logged: this.props.base.logged}, () => console.log('re'));
		}
	}

	toggle() {
		this.setState({
			isOpen: !this.state.isOpen
		});
	}

	capitalize = function(str) {
		return str.charAt(0).toUpperCase() + str.slice(1);
	};

	editFilter(filter) {
		if (filter in this.state.filters) {
			this.setState({
				filters: this.state.slice(this.state.filters.findIndex(filter))
			});
		} else {
			this.setState({
				filters: this.state.filter.push(filter)
			})
		}
	}

	changeChild(child) {
		if (typeof child === 'string' && child === 'general') {
			axios.post('/get_data').then(response => {
				const data = response.data.map(obj => {
					return {name: this.capitalize(obj.key.toLowerCase()), value: obj.value};
				});
				this.setState({selectedChild: 'General', isOpen: false, childData: data});
			}).catch(err => {
				console.log(err);
			})
		} else {
			this.getChildData(child)
		}
	}

	getChildData(child) {
		axios.post('/get_data', {
			discordId: child.discordId
		}).then(response => {
			const data = response.data.map(obj => {
				return {name: this.capitalize(obj.key.toLowerCase()), value: obj.value};
			});
			this.setState({selectedChild: child.name, isOpen: false, childData: data});
		}).catch(err => {
			console.log(err);
		})
	}

	render() {
		let i = 0;
		if (this.state.logged) {
			return (
				<div>
					<div>
						<ButtonDropdown isOpen={this.state.isOpen} toggle={this.toggle} size="lg">
							<DropdownToggle color="info">
								{this.state.selectedChild ? this.state.selectedChild : displayContent(this.state.lang, i++, 'report')}
							</DropdownToggle>
							<DropdownMenu className="drop btn">
								{
									this.state.childrens.map((d, idx) => {
										return (
											<div>
												<div onClick={() => {this.changeChild(d)}}>{d.name}</div>
											</div>
										)
									})
								}
								<div>
									<div onClick={() => {this.changeChild('general')}}>General</div>
								</div>
							</DropdownMenu>
						</ButtonDropdown>
					</div> <br/>

					<div className="left-filter">
						<span>{displayContent(this.state.lang, i++, 'report')}</span> <br/> <br/>
						<div className="filter-align">
							<input type="checkbox"/> Discord <br/>
							<input type="checkbox"/> {displayContent(this.state.lang, i++, 'report')} <br/>
						</div>
					</div>

					<div style={{width: '50%'}} className="btn">
						{
							this.state.childData.map((d, idx) => {
								if (d.name === 'Safe') {
									return (
										<div style={{height: 32, margin: 10}}>
											<text style={{float: "left", marginRight: 10, width: 150, textAlign: 'right'}}>{displayContent(this.state.lang, idx + 1, 'report')}</text><ProgressBar style={{height: 24, fontSize: 15}} animated striped variant="success" now={Math.round(d.value)} label={`${Math.round(d.value)}%`}/>
										</div>
									);
								} else {
									var color;
									if (d.value <= 33)
										color = "info";
									else if (d.value <= 66)
										color = "warning";
									else
										color = "danger";
									return (
										<div style={{height: 32, margin: 10}}>
											<text style={{float: "left", marginRight: 10, width: 150, textAlign: 'right'}}>{displayContent(this.state.lang, idx + 1, 'report')}</text><ProgressBar style={{height: 24, fontSize: 15}} animated striped variant={color} now={Math.round(d.value)} label={`${Math.round(d.value)}%`}/>
										</div>
									);
								}
							})
						}
					</div>
					<Summary lang={this.state.lang} child={this.state.selectedChild} safe={this.state.childData.length > 0 ? Math.round(this.state.childData.find(c => c.name === 'Safe').value) : -1}/>
				</div>
			);
		} else {
			return (
				<div>
					<Unauthorized/>
				</div>
			)
		}
	}
}

class Summary extends Component {

    componentDidMount() {
        console.log("CACA", this.props.safe);
    }

    render() {
    	let i = 0;
	    if (this.props.child  && this.props.safe > 0) {
	        return (
                <div>
                    <h5> {displayContent(this.props.lang, i++, 'summary')} {this.props.child} {displayContent(this.props.lang, i++, 'summary')} {this.props.safe} {displayContent(this.props.lang, i++, 'summary')}</h5>
                    <br/>
	                {this.props.safe >= 0 && this.props.safe < 33 && <img src={bad} alt="bad"/>}
	                {this.props.safe >= 33 && this.props.safe < 66 && <img src={neutral} alt="neutral"/>}
	                {this.props.safe >= 66 && this.props.safe <= 100 && <img src={good} alt="good"/>}
                </div>
            )
        } else if (this.props.safe === -1 && this.props.child) {
	    	i = 3;
	    	return (
	    		<div>
					<h5>{displayContent(this.props.lang, i++, 'summary')} {this.props.child} {displayContent(this.props.lang, i++, 'summary')}</h5>
			    </div>
		    )
	    } else {
	        return null
        }
	}
}

export default Report;