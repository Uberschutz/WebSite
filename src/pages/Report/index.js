import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import * as actions from '../../actions'
import Report from './Report'

const mapStateToProps = state => ({
	base: state.base,
})

const mapDispatchToProps = dispatch => ({

})

const ReportContainer = connect(mapStateToProps, mapDispatchToProps)(Report)

export default ReportContainer