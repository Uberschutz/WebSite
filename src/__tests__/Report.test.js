import React from 'react';
import ReactDOM from 'react-dom';
import Report from '../pages/Report/Report';
//import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

it('renders Report without crashing (not connected) in any language', () => {
	const div = document.createElement('div');
	ReactDOM.render(<Report/>, div);
	expect(div).toMatchSnapshot();
	ReactDOM.unmountComponentAtNode(div);
});

it('render Report being logged in french', () => {
	const div = document.createElement('div');
	const paramsComponent = shallow(<Report/>);
	paramsComponent.setProps({base: { language: 'fr'}});
	paramsComponent.setState({ lang: 'fr', logged: true });
	ReactDOM.render(paramsComponent, div);
	expect(div).toMatchSnapshot();
	ReactDOM.unmountComponentAtNode(div);
});

it('render Report being logged in english', function () {
	const div = document.createElement('div');
	const paramsComponent = shallow(<Report/>);
	//paramsComponent.setProps({base: { logged: true, lang: 'fr' }});
	paramsComponent.setState({ lang: 'en', logged: true });
	ReactDOM.render(paramsComponent, div);
	expect(div).toMatchSnapshot();
	ReactDOM.unmountComponentAtNode(div);
});
