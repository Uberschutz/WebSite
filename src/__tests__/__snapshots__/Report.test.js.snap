// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Report page rendering render Report being logged in english 1`] = `
<div>
  <div>
    <div>
      <div
        class="mrg-begin btn-group btn-group-lg"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle btn btn-info"
          type="button"
        >
          Kids
        </button>
        <div
          aria-hidden="true"
          class="drop btn dropdown-menu"
          role="menu"
          tabindex="-1"
        >
          <div>
            <div
              data-child="general"
            >
              Example
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="left-filter"
    >
      <span>
        Filters:
      </span>
       
      <br />
       
      <br />
      <div
        class="filter-align"
      >
        <input
          data-filter="Discord"
          type="checkbox"
        />
         Discord 
        <br />
        <input
          data-filter="plugin"
          type="checkbox"
        />
         
        Internet browsing
         
        <br />
      </div>
    </div>
    <div
      class="btn content-right"
      style="width: 70%;"
    >
      <div
        class="summary"
      />
    </div>
  </div>
</div>
`;

exports[`Testing the Report page rendering render Report being logged in french 1`] = `
<div>
  <div>
    <div>
      <div
        class="mrg-begin btn-group btn-group-lg"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle btn btn-info"
          type="button"
        >
          Enfants
        </button>
        <div
          aria-hidden="true"
          class="drop btn dropdown-menu"
          role="menu"
          tabindex="-1"
        >
          <div>
            <div
              data-child="general"
            >
              Example
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="left-filter"
    >
      <span>
        Filtres :
      </span>
       
      <br />
       
      <br />
      <div
        class="filter-align"
      >
        <input
          data-filter="Discord"
          type="checkbox"
        />
         Discord 
        <br />
        <input
          data-filter="plugin"
          type="checkbox"
        />
         
        Navigation internet
         
        <br />
      </div>
    </div>
    <div
      class="btn content-right"
      style="width: 70%;"
    >
      <div
        class="summary"
      />
    </div>
  </div>
</div>
`;

exports[`Testing the Report page rendering render Report not being logged in english 1`] = `
<div>
  <div>
    <div
      class="responsive-image"
    >
      <h1
        class="text-xl-center forbidden"
      >
        403 Accès interdit
      </h1>
      <br />
      <img
        alt="forbidden"
        class="responsive-image"
        src="fogg-no-comments.png"
      />
    </div>
  </div>
</div>
`;

exports[`Testing the Report page rendering render Report not being logged in french 1`] = `
<div>
  <div>
    <div
      class="responsive-image"
    >
      <h1
        class="text-xl-center forbidden"
      >
        403 Accès interdit
      </h1>
      <br />
      <img
        alt="forbidden"
        class="responsive-image"
        src="fogg-no-comments.png"
      />
    </div>
  </div>
</div>
`;

exports[`Testing the Report page rendering renders Report without crashing (not connected) in any language 1`] = `
<div>
  <div>
    <div
      class="responsive-image"
    >
      <h1
        class="text-xl-center forbidden"
      >
        403 Accès interdit
      </h1>
      <br />
      <img
        alt="forbidden"
        class="responsive-image"
        src="fogg-no-comments.png"
      />
    </div>
  </div>
</div>
`;
